### PR TITLE
Fix due date rendering the wrong date in issue

### LINKED ISSUE #21999: Due is shown incorrectly
### Description

The issue is here: https://try.gitea.io/pan_dunski/test/issues/1

I've chosen due as 2022-12-02 but after save it is converted to 2022-12-03

My timezone is CEST

### Gitea Version

1.18.0+rc1

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

image

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

I'm using official chart with the following values:

```yaml
image:
  repository: gitea/gitea
  pullPolicy: Always
  tag: "1.18.0-rc1"
```

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by yardenshoham**
Previously, the last minute of the chosen date caused bad timezone rendering.

For example, I chose January 4th, 2023.

### Before
```html
<time data-format="date" datetime="Wed, 04 Jan 2023 23:59:59 +0000">January 5, 2023</time>
```

### After
```html
<time data-format="date" datetime="2023-01-04">January 4, 2023</time>
```

---

Closes #21999

### CODE REVIEW

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 1]
```diff
@@ -413,7 +413,7 @@
 					<div class="df sb ac">
 						<div class="due-date tooltip {{if .Issue.IsOverdue}}text red{{end}}" {{if .Issue.IsOverdue}}data-content="{{.locale.Tr "repo.issues.due_date_overdue"}}"{{end}}>
 							{{svg "octicon-calendar" 16 "mr-3"}}
-							<time data-format="date" datetime="{{.Issue.DeadlineUnix.FormatLong}}">{{.Issue.DeadlineUnix.FormatDate}}</time>
```
Comment (delvh) [inline templates/repo/issue/view_content/sidebar.tmpl:416]: I am not quite sure why `FormatLong` and `FormatDate` should produce different results?

**Review by delvh — APPROVED** [review]

File: templates/repo/issue/view_content/sidebar.tmpl (hunk 1 above)
Comment (yardenshoham) [reply templates/repo/issue/view_content/sidebar.tmpl:416]: Looks like `FormatDate` gives only the date, without HH:MM:SS (see example in PR description)

Comment (delvh) [reply templates/repo/issue/view_content/sidebar.tmpl:416]: Yeah, but both should still produce the correct date (`2023-01-04`), right?
So why should suddenly a day be added?
That's what I don't understand…

Comment (yardenshoham) [reply templates/repo/issue/view_content/sidebar.tmpl:416]: `FormatLong` gives `Wed, 04 Jan 2023 23:59:59 +0000` so I think that when the date is localized in javascript it is either:
1. rounded
2. timezoned

It's just a guess though

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**